### PR TITLE
WZ-147:-Added a button to copy the configuration from Another Store

### PR DESCRIPTION
**Pull request by smit-kandiya**

### CODE REVIEW

File: Model/Admin/Source/GetStoreNames.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Wizzy\Search\Model\Admin\Source;
+
+use Magento\Store\Model\StoreManagerInterface;
+
+class GetStoreNames
```
Comment (alok0590) [inline Model/Admin/Source/GetStoreNames.php:7]: ONly StoreNames

File: Model/Admin/Source/GetStoreNames.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        $this->storeManager = $storeManager;
+    }
+
+    public function toOptionArray()
+    {
+        $storeNames = [
+            [
+                "value" => '',
+                "label" => 'select'
```
Comment (alok0590) [inline Model/Admin/Source/GetStoreNames.php:21]: Select

File: Model/Admin/Source/GetStoreNames.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            ]
+        ];
+
+        foreach ($this->storeManager->getStores() as $store) {
```
Comment (alok0590) [inline Model/Admin/Source/GetStoreNames.php:25]: We already have this logic inside a Store Manager. Which we use to fetch list of stores in products indexer. Please use the same.

File: Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            StoreAdvancedConfig::INCLUDE_CUSTOM_CSS,
+            StoreAdvancedConfig::TEMPLATE_ATTRIBUTES
+        ];
+
+        foreach ($configPath as $path) {
+            $oldConfig[$path] = $this->configManager->getStoreConfig($path, $fromStoreId);
+        }
+
+        foreach ($configPath as $path) {
+            $newConfig[$path] = $this->configManager->getStoreConfig($path, $toStoreId);
```
Comment (alok0590) [inline Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php:115]: If we're not using it anywhere please remove it.

File: Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        $this->configManager = $configManager;
+        $this->storeCopyConfig = $storeCopyConfig;
+    }
+
+    public function copyConfiguration()
+    {
+        $fromStoreId = $this->storeCopyConfig->getCopyConfigFrom();
+        $toStoreId = $this->request->getParam('store');
+        $oldConfig = [];
+        $newConfig = [];
```
Comment (alok0590) [inline Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php:41]: Remove if not required.

File: Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        foreach ($oldConfig as $path => $value) {
+            $this->_configWriter->save(
+                $path,
+                $value,
+                $scopeType,
+                $toStoreId
+            );
+        }
+
+        $cacheType = 'config';
```
Comment (alok0590) [inline Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php:126]: Do we have any const available? If yes, use it please.

File: etc/adminhtml/menu.xml [hunk 7]
```diff
@@ -81,5 +81,14 @@
             action="wizzy_search/sync/skippedentities"
             resource="Wizzy_Search::main"
         />
+         <add
```
Comment (alok0590) [inline etc/adminhtml/menu.xml:84]: Remove this from Menu please.

File: etc/adminhtml/system.xml [hunk 8]
```diff
@@ -886,6 +886,23 @@
...
+            <tab>wizzy_config</tab>
+            <resource>Wizzy_Search::wizzy_config</resource>
+            <group id="copy_configuration" translate="label" type="text" sortOrder="10" showInDefault="0" showInWebsite="0" showInStore="1">
+                <label>Copy Configuration From</label>
+                <field id="from_store" translate="label" type="select" sortOrder="20" showInDefault="0" showInWebsite="0" showInStore="1">
+                    <label>Store</label>
+                    <source_model>Wizzy\Search\Model\Admin\Source\GetStoreNames</source_model>
+                    <comment>
+                        <![CDATA[
+                            Select store name from where you want to copy configurations.
```
Comment (alok0590) [inline etc/adminhtml/system.xml:901]: Select store from where you want to copy configurations.

**Review by alok0590 — CHANGES_REQUESTED** [review]

File: etc/adminhtml/menu.xml (hunk 7 above)
Comment (smit-kandiya) [reply etc/adminhtml/menu.xml:84]: Can i put it under 'Advanced Configuration' ?

File: Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php (hunk 6 above)
Comment (smit-kandiya) [reply Model/Observer/AdminConfigs/CopyConfigurationOptionUpdated.php:126]: I think there is no any Const for cacheType , we have to pass config type as string only.

File: Model/Admin/Source/StoreNames.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    public function toOptionArray()
+    {
+        $storeNames = [
+            [
+                "value" => '',
+                "label" => 'Select'
+            ]
+        ];
+        $storeName = $this->_storeManager->getActivateWizzyStoreNames();
```
Comment (alok0590) [inline Model/Admin/Source/StoreNames.php:28]:
```suggestion
        $storeNames = $this->_storeManager->getActivateWizzyStoreNames();
```

**Review by alok0590 — CHANGES_REQUESTED** [review]

**Review by alok0590 — APPROVED** [review]

**Pull request merged**